### PR TITLE
fix: start button in job card does not add new timelog on adding empl…

### PR DESCRIPTION
**Pull request by hitenshKharva**
Ref: [Asana Task](https://app.asana.com/0/1199082161983365/1199967685167812)

Referred upstream erpnext for the fix : https://github.com/frappe/erpnext/pull/23226

https://user-images.githubusercontent.com/56474017/110201928-b78da780-7e8b-11eb-84dd-74c01d459e8d.mp4

### CODE REVIEW

File: erpnext/manufacturing/doctype/job_card/job_card.js [hunk 1]
```diff
@@ -70,9 +70,10 @@ frappe.ui.form.on('Job Card', {
 						fieldname: 'employee'}, d => {
 						if (d.employee) {
 							frm.set_value("employee", d.employee);
+						} else {
+							frm.events.start_job(frm);
```
Comment (hrwX) [inline erpnext/manufacturing/doctype/job_card/job_card.js:74]:
```suggestion
						}
						frm.events.start_job(frm);
```

Row should be added even after employee is set

Comment (hitenshKharva) [reply erpnext/manufacturing/doctype/job_card/job_card.js:74]: Earlier a row was added when employee is set, which created double entries for stock quantities on completion. 
The above fix is as per the erpnext upstream fix for the issue.

**Comment by hitenshKharva** [comment]
Changes as requested.


https://user-images.githubusercontent.com/56474017/110285361-29d7c680-8009-11eb-978f-9a16454b8e59.mp4

**Review by rvpasari — DISMISSED** [review]

File: erpnext/manufacturing/doctype/job_card/job_card.js [hunk 2]
```diff
@@ -39,7 +39,7 @@ frappe.ui.form.on('Job Card', {
 		}
 
 		if (frm.doc.docstatus == 0 && (frm.doc.for_quantity > frm.doc.total_completed_qty || !frm.doc.for_quantity)
-			&& (!frm.doc.items.length || frm.doc.for_quantity == frm.doc.transferred_qty)) {
+			&& (frm.doc.items || !frm.doc.items.length || frm.doc.for_quantity == frm.doc.transferred_qty)) {
```
Comment (sahil28297) [inline erpnext/manufacturing/doctype/job_card/job_card.js]:
```suggestion
			&& (frm.doc.for_quantity == frm.doc.transferred_qty)) {
```
your condition currently suggests that that condition will pass if "items are there" and even if "theres no length for items", then why do you need item condition if it works for wither of the case?

**Review by sahil28297 — CHANGES_REQUESTED** [review]

**Comment by sahil28297** [comment]
@hitenshKharva if you are pulling something from upstream, please ensure that you also mention that PR in your description.

**Comment by hitenshKharva** [comment]
@sahil28297  Sure.

Upstream PR: https://github.com/frappe/erpnext/pull/23226

**Pull request merged**